src/libData/BlockData/BlockHeader/TxBlockHeader.cpp

```diff
@@ -218,6 +218,18 @@ const BlockHash& TxBlockHeader::GetDSBlockHeader() const
     return m_dsBlockHeader;
 }
 
+BlockHash TxBlockHeader::GetMyHash() const
+{
+    SHA2<HASH_TYPE::HASH_VARIANT_256> sha2;
+    std::vector<unsigned char> vec;
+    Serialize(vec, 0);
+    sha2.Update(vec);
+    const std::vector<unsigned char>& resVec = sha2.Finalize();
+    BlockHash blockHash;
+    std::copy(resVec.begin(), resVec.end(), blockHash.asArray().begin());
+    return blockHash;
+}
+
```

Comment:
Since GetMyHash() logic are identical between Ds and Tx, it would be nice to put onto BlockHeaderBase class.